From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Wed, 14 Aug 2024 12:16:24 +0800
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7107eabf02d9d9b69d97c978ba421ad52031ec3f..08de792cd0090305d0c2cf1271330655992d2125 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +25,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // SkyblockPurpur start
+    /**
+     * The brand id for SkyblockPurpur
+     */
+    Key BRAND_SKYBLOCKPURPUR_ID = Key.key("skyblockpurpurmc", "skyblockpurpur");
+    // SkyblockPurpur end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
