From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Wed, 14 Aug 2024 12:16:24 +0800
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7196594e07af19a14c320d77df893978525fe386..9bab10193777029533bc458a75f5760661f3dd62 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -33,6 +33,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // SkyblockPurpur start
+    /**
+     * The brand id for SkyblockPurpur
+     */
+    Key BRAND_SKYBLOCKPURPUR_ID = Key.key("skyblockpurpurmc", "skyblockpurpur");
+    // SkyblockPurpur end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
