From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Thu, 29 Aug 2024 14:08:57 +0800
Subject: [PATCH] custom structure register


diff --git a/src/main/java/hk/catgirl/skyblockpurpur/CustomStructureRegister.java b/src/main/java/hk/catgirl/skyblockpurpur/CustomStructureRegister.java
new file mode 100644
index 0000000000000000000000000000000000000000..10aa755e6786573bcfba666394774bf03aa07b20
--- /dev/null
+++ b/src/main/java/hk/catgirl/skyblockpurpur/CustomStructureRegister.java
@@ -0,0 +1,8 @@
+package hk.catgirl.skyblockpurpur;
+
+public interface CustomStructureRegister {
+    public CustomStructureRegister INSTANCE = null;
+
+    public String getRegionStructure(int region_x, int region_z);
+    public String getRegionStructure(String dimension, int region_x, int region_z);
+}
