From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Thu, 29 Aug 2024 14:08:57 +0800
Subject: [PATCH] custom structure register


diff --git a/src/main/java/hk/catgirl/skyblockpurpur/CustomStructureRegister.java b/src/main/java/hk/catgirl/skyblockpurpur/CustomStructureRegister.java
new file mode 100644
index 0000000000000000000000000000000000000000..746b70b44d2e23618dbd88af8382146d5c597f3e
--- /dev/null
+++ b/src/main/java/hk/catgirl/skyblockpurpur/CustomStructureRegister.java
@@ -0,0 +1,12 @@
+package hk.catgirl.skyblockpurpur;
+
+import io.papermc.paper.math.BlockPosition;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface CustomStructureRegister {
+    public @Nullable CustomStructureRegister INSTANCE = null;
+
+    public @NotNull String getRegionStructure(@NotNull BlockPosition position);
+    public @NotNull String getRegionStructure(@NotNull String dimension, @NotNull BlockPosition position);
+}
