From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Thu, 29 Aug 2024 14:08:57 +0800
Subject: [PATCH] custom structure register


diff --git a/src/main/java/hk/catgirl/skyblockpurpur/events/MobSpawnStructureEvent.java b/src/main/java/hk/catgirl/skyblockpurpur/events/MobSpawnStructureEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..bf20a6e08be65a3a61af50789421ba99eda2eccd
--- /dev/null
+++ b/src/main/java/hk/catgirl/skyblockpurpur/events/MobSpawnStructureEvent.java
@@ -0,0 +1,40 @@
+package hk.catgirl.skyblockpurpur.events;
+
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class MobSpawnStructureEvent extends Event {
+    private static final HandlerList handlers = new HandlerList();
+    private final @NotNull Location position;
+
+    public @Nullable String structure = null;
+
+    public MobSpawnStructureEvent(@NotNull Location position) {
+        this.position = position;
+    }
+
+    public @NotNull Location getPosition() {
+        return position;
+    }
+
+    public void setStructure(@Nullable String structure) {
+        this.structure = structure;
+    }
+
+    public @Nullable String getStructure() {
+        return structure;
+    }
+
+    @Override
+    public @NotNull HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static @NotNull HandlerList getHandlerList() {
+        return handlers;
+    }
+
+}
