From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Wed, 14 Aug 2024 12:32:28 +0800
Subject: [PATCH] Relocate end portal destination


diff --git a/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java b/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java
index 28d8c8a879badc97cd1168e1847e2fafdc40e4bd..0bec147c270cf9a3baf8cce31d3d66e4c79a0957 100644
--- a/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java
@@ -101,7 +101,7 @@ public class EndPortalBlock extends BaseEntityBlock implements Portal {
             return null; // Paper - keep previous behavior of not firing PlayerTeleportEvent if the target world doesn't exist
         } else {
             boolean flag = resourcekey == Level.END;
-            BlockPos blockposition1 = flag ? ServerLevel.END_SPAWN_POINT : worldserver1.getSharedSpawnPos();
+            BlockPos blockposition1 = new BlockPos(((pos.getX() << 9) >> 9) + 256, pos.getY(), ((pos.getZ() << 9) >> 9) + 256);
             Vec3 vec3d = blockposition1.getBottomCenter();
             float f = entity.getYRot();
 
