From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Wed, 14 Aug 2024 12:32:28 +0800
Subject: [PATCH] Relocate end portal destination


diff --git a/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java b/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java
index 9c11a615fbfd4fcaff5df2a15d1b8578897ab2b4..904a0a70c39e6067663acff6ceeea9ea1ecb906a 100644
--- a/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/EndPortalBlock.java
@@ -100,7 +100,7 @@ public class EndPortalBlock extends BaseEntityBlock implements Portal {
             return new DimensionTransition(PlayerTeleportEvent.TeleportCause.END_PORTAL); // CraftBukkit- always fire event in case plugins wish to change it
         } else {
             boolean flag = resourcekey == Level.END;
-            BlockPos blockposition1 = flag ? ServerLevel.END_SPAWN_POINT : worldserver1.getSharedSpawnPos();
+            BlockPos blockposition1 = new BlockPos(pos.getX(), pos.getY(), pos.getZ());
             Vec3 vec3d = blockposition1.getBottomCenter();
             float f = entity.getYRot();
 
